src/wwwroot/template-editor/adapterService.js

```diff
@@ -354,17 +375,28 @@
      */
     function getCampaign(id, useEditorAsTemplate) {
       traceAdapterCall(getCampaign, arguments);
-      var campaign = {
-        id: id,
-        type: 'campaign',
-        name: 'Campaign ' + id,
-        attributes: {},
-        innerHTML: '',
-        children: []
-      };
-      return $q.resolve({ data: campaign });
+      return $http.get(`${RELAY_CONFIG.baseUrl}/accounts/${loginSession.accountName}/template/${id}`, {
+        headers: {
+          'Authorization': 'Bearer '+ apiToken
+        }
+      }).then(function(response) {
+        var campaign = {
+          id: id,
+          type: 'campaign',
+          name: 'Campaign ' + id,
+          attributes: {},
+          innerHTML: '',
+          children: []
+        };
+        return { data: campaign };
+      },
+      function(error) {
+        var errorDetail = error.data && error.data.detail || "Unexpected error";
+        showModalAndRedirect("/#/templates", errorDetail);
+        return $q.reject();
```

Comment:
I am not sure, but in my opinion, this behavior is not correct:

```js
function(error) {
  // . . .
  showModalAndRedirect("/#/templates", errorDetail);
  return $q.reject();
}
```

It is returning the error (`return $q.reject()`), to continue the normal error flow, in parallel of showing the dialog (`showModalAndRedirect("/#/templates", errorDetail);`). 

In my opinion, we should wait for the user closing the dialog before continue with the error flow, the code sould be something like:

```js
function(error) {
  // . . .
  return showModalAndRedirect("/#/templates", errorDetail).then(function() {
    return $q.reject(error);
  });
}
```

Notice that you are also losing the original error, I think that the right way is returning `return $q.reject(error);`, but please research for it.

Reply:
I will do it, thanks andres